bolt5: Test ignorance of invalid tx

ariard opened the pull request:
Hmmmm in fact a too zealous /useless test, because all invalid txn should be discarded at chain layer.
Or we may want more tx validity checks in block_connected if ChainWatchInterface isn't a full node ? (but seems anyway a layer violation)

TheBlueMatt commented:
Hmmmmm, indeed, I'm not sure what the value is here. Testing that a related-but-different-txid tx confirming doesn't result in the channel confirming is fine, but the "SHOULD ignore invalid transactions" recommendation Rationale seems to be more focused on either unconfirmed transactions or an invalid block. I suppose we could double-check things with libbitcoinconsensus before acting on them, but mining an invalid block to get us to close a channel I'm not too worried about since we don't even lose money for that (they'd need HTLC_FAIL_ANTI_REORG_DELAY blocks to get us to fail an HTLC, which hopefully is way more valuable than any HTLCs we accept).

Checking that a random transaction doesn't result in a channel failure is hopefully kinda unnecessary.

ariard commented:
Okay if you assess that vector attacks with invalid txn are impracticable/uneconomical, I think it's better to avoid performance hit with double-check.
Agree that first part of test on funding tx isn't targeted by Rationale but thought it was interesting to test and quite near in scope. Deleted unnecessary second part, renamed test as test_invalid_funding_tx, but if you deem it as superfluous just close it

TheBlueMatt commented:
Yea, I think this test as-is isn't so useful. That said, for a related test that I think would be useful, we should check that a transaction at 1 confirmation that is apparently an HTLC-Success claim but has the wrong preimage results in no action (and we're still willing to fail-back the HTLC after HTLC_FAIL_ANTI_REORG_DELAY confirmations if we don't learn a preimage).

TheBlueMatt commented:
Closing, added issue for the new test idea at #293.
